### PR TITLE
Add PInvoke/SizeParamIndex tests

### PR DESCRIPTION
**Pull request by jiangzeng**

### CODE REVIEW

File: tests/src/Interop/PInvoke/SizeParamIndex/PInvoke/PassingByOut/PInvokePassingByOutNative.cpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// PInvokePassingByOutNative.cpp : Defines the entry point for the DLL application.
+//
+#include <xplatform.h>
+#include "../helper.h"
```
Comment (jiangzeng) [inline tests/src/Interop/PInvoke/SizeParamIndex/PInvoke/PassingByOut/PInvokePassingByOutNative.cpp]: @AaronRobinsonMSFT @luqunl Do you have any suggestion about how to include files with relative path like this?
Is this OK?

Comment (AaronRobinsonMSFT) [reply tests/src/Interop/PInvoke/SizeParamIndex/PInvoke/PassingByOut/PInvokePassingByOutNative.cpp]: If this is indeed the location of the header, this is fine. Are you concerned about something specific?

Comment (jiangzeng) [reply tests/src/Interop/PInvoke/SizeParamIndex/PInvoke/PassingByOut/PInvokePassingByOutNative.cpp]: @AaronRobinsonMSFT There will be compile warnings about using the relative path "../" to include a file. Just want to make sure if it's fine to use the relative path here

Comment (AaronRobinsonMSFT) [reply tests/src/Interop/PInvoke/SizeParamIndex/PInvoke/PassingByOut/PInvokePassingByOutNative.cpp]: ugh.... What is the warning? You might want to try `..\helper.cpp`. If that is still a warning please add `..` to the include path in the `CMakeLists.txt` for the relevant projects.

Comment (jiangzeng) [reply tests/src/Interop/PInvoke/SizeParamIndex/PInvoke/PassingByOut/PInvokePassingByOutNative.cpp]: The warning is:
`warning C4464: relative include path contains '..'`
I will try to include in the CMakeLists.txt to see.

Comment (jiangzeng) [reply tests/src/Interop/PInvoke/SizeParamIndex/PInvoke/PassingByOut/PInvokePassingByOutNative.cpp]: This is fixed by adding `include_directories("..")` to CMakeList.txt and then `#include "helper.h"`

**Comment by jkoritzinsky** [comment]
@janvorli @adityam We're having an issue on this PR with exceptions off-Windows.

When an `OverflowException` exception is being thrown by the runtime (via a `COMPlusThrow` call) during the setup for a reverse P/Invoke, called from managed code (so transitioning back into managed from a managed-> native call-stack), the exception seems to ignore two `catch (Exception) {}` clauses in its stack trace that should catch the exception. It does work on Windows BTW. We did step through it.

cc: @AaronRobinsonMSFT

**Comment by jkoritzinsky** [comment]
@dotnet-bot test this please

**Comment by jkoritzinsky** [comment]
@adityamandaleeka

**Comment by jkoritzinsky** [comment]
@adityamandaleeka @AaronRobinsonMSFT update on trying a more specific catch clause. Trying to catch the `OverflowException` with a `catch(OverflowException){}` block also does not work.

**Comment by adityamandaleeka** [comment]
@jkotas I assume nothing has changed since you and I talked about this scenario back in 2015? 😄

**Comment by jkotas** [comment]
Correct, exception interop is supported on Windows only.

**Comment by AaronRobinsonMSFT** [comment]
> exception interop is supported on Windows only.

This sounds very familiar, I know I have talked with @adityamandaleeka about this before. Just so @jkoritzinsky and I are clear on this point. There is _no_ support for exceptions crossing the interop boundary right? I would assume this works on Windows due to SEH.

**Comment by jkotas** [comment]
Right. https://github.com/dotnet/docs/issues/6629#issuecomment-408722262 tracks adding this to the official documentation.

**Comment by jkoritzinsky** [comment]
I've disabled the throwing test off-Windows.

**Pull request merged**